components/provisioner/internal/gardener/auditlogging_test.go

```diff
@@ -0,0 +1,175 @@
+package gardener
+
+import (
+	"path/filepath"
+	"testing"
+
+	gardener_types "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestAuditLogConfigurator_CanEnableAuditLogsForShoot(t *testing.T) {
+	t.Run("should return true when seedName and auditLogTenantConfigPath are not empty", func(t *testing.T) {
+		//given
+		auditLogConfigurator := NewAuditLogConfigurator("/path")
+		seedName := "az-eu3"
+
+		//when
+		enable := auditLogConfigurator.CanEnableAuditLogsForShoot(seedName)
+
+		//then
+		assert.True(t, enable)
+	})
+
+	t.Run("should return false when auditLogTenantConfigPath is empty", func(t *testing.T) {
+		//given
+		auditLogConfigurator := NewAuditLogConfigurator("")
+		seedName := "az-eu3"
+
+		//when
+		enable := auditLogConfigurator.CanEnableAuditLogsForShoot(seedName)
+
+		//then
+		assert.False(t, enable)
+	})
+
+	t.Run("should return false when seedName is empty", func(t *testing.T) {
+		//given
+		auditLogConfigurator := NewAuditLogConfigurator("/path")
+		seedName := ""
+
+		//when
+		enable := auditLogConfigurator.CanEnableAuditLogsForShoot(seedName)
+
+		//then
+		assert.False(t, enable)
+	})
+}
+
+func TestAuditLogConfigurator_SetAuditLogAnnotation(t *testing.T) {
```

Comment:
There could be also a test when the shoot is already annotated so the function will return `false, nil`, right?